lib/mixpanel/queue.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Mixpanel.Queue do
+  def new(limit), do: {0, limit, [], []}
+
+  def push({len, limit, _, _}, _item) when len >= limit do
+    :dropped
+  end
+
+  def push({len, limit, tail, head}, item) do
+    {:ok, {len + 1, limit, [item | tail], head}}
+  end
+
+  def take({len, limit, [], head}, max) do
+    case Enum.split(head, max) do
+      {result, []} ->
+        {:ok, result, {0, limit, [], []}}
+
+      {result, new_head} ->
+        {:ok, result, {len - max, limit, [], new_head}}
```

Comment:
what if `max > len`: `Enum.split/2` handles it correctly, but `len - max` doesn't.

Reply:
Also, interesting results when `max < 0` :)

Reply:
Yeah, I'll add a guard for max to be >0